### PR TITLE
Activate PSFMap for fitting

### PR DESCRIPTION
**Pull request by AtreyeeS**
**Description**

This pull request activates `PSFMap` on the MapEvaluator. Similar to what is done in #2587 , the option to keep PSFKernel is still on. Once this PR is merged, I will deactivate the kernels and adapt the notebooks.

**Dear reviewer**

One specific doubt I have is what should be the `max_radius` in 
`self.psf = psf.get_psf_kernel(self.model.position, geom=exposure.geom, max_radius=0.8 * u.deg)`
Having it as a parameter will be extremely confusing for the users. Currently I am passing a large enough value which should hopefully suffice for most cases.

I did not adapt the tests in `analysis` to avoid conflicts with the changes @Bultako might be making.

### CODE REVIEW

File: gammapy/cube/fit.py [hunk 1]
```diff
@@ -1418,20 +1418,23 @@ def update(self, exposure, psf, edisp, geom):
...
             e_reco = geom.get_axis_by_name("energy").edges
             self.edisp = edisp.get_energy_dispersion(self.model.position, e_reco=e_reco)
         else:
             self.edisp = edisp
 
-        # TODO: lookup correct PSF for this component
-        self.psf = psf  # .get_psf_kernel(self.model.position, geom=exposure.geom)
+        if isinstance(psf, PSFMap):
+            self.psf = psf.get_psf_kernel(
+                self.model.position, geom=exposure.geom, max_radius=0.8 * u.deg
```
Comment (adonath) [inline gammapy/cube/fit.py]: I would propose to modify `.get_psf_kernel()` and just take the maximum of `table_psf.rad` as a default maximum radius (`if max_radius is  None`). This is more consistent then having a separate hard-coded value here...

**Comment by AtreyeeS** [comment]
Thanks @adonath ! I have made the change

**Comment by adonath** [comment]
Thanks @AtreyeeS! I've just restarted the remaining failing Travis builds, once those are green Ill merge...

**Comment by adonath** [comment]
Ok, it seems `gammapy download` is broken...the Travis-CI fails are unrelated. The tests pass locally for me. I'll go ahead and merge now...

**Pull request merged**